Dockerfile

```diff
@@ -10,17 +10,55 @@ SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 USER root
 
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update --yes && \
+    # - `apt-get upgrade` is run to patch known vulnerabilities in system packages
+    #   as the Ubuntu base image is rebuilt too seldom sometimes (less than once a month)
+    apt-get upgrade --yes && \
+    apt-get install --yes --no-install-recommends build-essential gfortran && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+USER ${NB_UID}
+RUN set -x && \
+    # Installing jupyter lab extensions in the base environment
+    mamba install --yes jupyterlab-git=0.44.0 \
+    mamba_gator=5.2.1 \
+    jupytext=1.16.1 \
+    jupyter-archive=3.4.0 \
+    jupyter-server-proxy=4.1.0 \
+    dask-labextension=6.2.0 \
+    ipywidgets=8.1.1 \
+    jupyterlab=3.6.7 \
+    jupyter_bokeh=3.0.7 && \
+    mamba clean --all -f -y
+
+USER root
 COPY marble_environment.yml /environment.yml
 
-# Installing jupyter lab extensions in the main environment
-RUN mamba install --yes jupyterlab-git jupyter_conda jupyter-archive jupyter-server-proxy dask-labextension
-# Creating a "Marble" environment
-RUN mamba env create -f /environment.yml
+USER ${NB_UID}
+RUN set -x && \
+    # Creating the "Marble" environment
+    mamba env create -f /environment.yml && \
+    mamba clean --all -f -y
 
-ENV PATH="/opt/conda/envs/marble/bin:$PATH"
+USER root
+# New kernels are installed in /usr/local/share/jupyter. Changing permissions of this location to be writable by the user.
+# This opens the possibility that in the jupyter environment a user can create a new environment and install a new kernel.
+RUN mkdir -p /usr/local/share/jupyter
+RUN fix-permissions /usr/local/share/jupyter
 
+USER ${NB_UID}
+ENV PATH="$MAMBA_ROOT_PREFIX/envs/marble/bin:$PATH"
 RUN python -m ipykernel install --name Marble
 
-USER ${NB_UID}
+# For import xesmf since esmf-8.4.0, see
+# https://github.com/conda-forge/esmf-feedstock/issues/91
+ENV ESMFMKFILE=/opt/conda/envs/marble/lib/esmf.mk
+
+USER root
```

Comment:
can this user switch and the fix-permissions be removed?

Reply:
Right, so as we talked about the other day, the change to user is needed to run `fix-permissions` to make the folders writable by the user.